app/models/user.rb

```diff
@@ -14,4 +16,35 @@ class User < ApplicationRecord
   has_secure_password
 
   before_save{email.downcase!}
+
+  # Returns the hash digest of the given string.
+  def self.digest string
+    cost =  if ActiveModel::SecurePassword.min_cost
+              BCrypt::Engine::MIN_COST
+            else
+              BCrypt::Engine.cost
+            end
+    BCrypt::Password.create string, cost: cost
+  end
+
+  class << self
+    # Returns a random token.
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
```suggestion
    update_column(:remember_digest, User.digest(remember_token))
```
cái này là hợp lý  hơn nhé, ko cần callback hay validate nào cả